test/request.js

```diff
@@ -175,6 +175,19 @@ describe('request', function () {
         .then(done, done);
     });
 
+    it('agent can be used to set default headers', function (done){
+      var agent = request.agent('https://httpbin.org');
+      agent.auth("user", "passwd");
+
+      agent
+        .get('/basic-auth/user/passwd')
+        .then(function (res){
+          res.should.have.status(200);
+          agent.close();
+        })
+        .then(done, done);
+    });
+
```

Comment:
Looks good... Lets also test an arbitrary auth header. Our tests with non-system users will be using a cookie and a bearer token, I'd like to be sure that they'll work. Monday :)

Reply:
Just to clarify this means not basic auth for this one? I made another test that sets an arbitrary default header (using .set as opposed to .auth) and verifies that it is sent. Is that what you had in mind?